### DISCUSSION TITLE
Arc

### DISCUSSION THREAD

**Discussion by Timerlen** [mozman/ezdxf#363] [Q&A]
Good morning
It is possible to make a point control on an arc just like a spline ?

**Comment by mozman** [comment]
I don't understand your question. Please, provide more details.

**Reply by Timerlen** [reply]
spline ma punkty kontrolne których ilość określa sie w parametrze segments . Czy jest jakaś metoda aby na łuku można nanieśc takie punkty ?
![11](https://user-images.githubusercontent.com/80266841/110820194-2c4b5200-828f-11eb-9052-56aafcc20a40.jpg)

My problem is with arches, I need to define the area needed for a detail,. Max_value_x min_value_x max_value_y min_value_y.
![13](https://user-images.githubusercontent.com/80266841/110821551-7a148a00-8290-11eb-92c4-ed137c7c3049.jpg)

In this case it is best to add a radius to the center and it will be ok. What will happen when the detail will look like this?

![12](https://user-images.githubusercontent.com/80266841/110821957-decfe480-8290-11eb-8a72-93de19ec0a99.jpg)
detail will be 2 x radius larger. I did various combinations with start_point ,end_point, StartAngle, EndAngle and there was always something wrong

**Comment by mozman** [comment]
The next version of ezdxf, v0.16b0 is already available on PyPI, has a new `bbox`  module, which can calculate the bounding box for DXF entities (see the [docs](https://ezdxf.mozman.at/docs/bbox.html)).

The calculation of bounding boxes for CIRCLE, ARC, ELLIPSE and SPLINE is based on curve flattening to a certain precision, because this works for all cases in 2D and 3D, but has the disadvantage, that is is slow and may not yield the perfect result. It is not guaranteed to find the exact extreme points of a curve but the result will be close.

```Python
from ezdxf import bbox

doc = ...  # your DXF document
entities = ...  # collect the entities
box = bbox.extents(entities)

# or use the whole model space
box = bbox.extents(doc.modelspace())

```

**Reply by Timerlen** [reply]
Thank you for your help. Documentation gives a nice perspective. The latest version ezdxf  is 0.15.2 on PyPI

**Reply by mozman** [reply]
https://pypi.org/project/ezdxf/0.16b0/
Beta versions have to be explicit installed.
